File: exercises/python/2_collections_and_ranges/evaluate.py
from unittest import TestCase
import re

from collections_and_ranges import binary_regex
from collections_and_ranges import hawaiian_regex
from collections_and_ranges import telephone_regex
from collections_and_ranges import six_plus_no_a_regex


class EvaluateBinary(TestCase):

    def test_zero(self):
        matches = re.fullmatch(binary_regex, '0')
        self.assertIsNotNone(
            matches,
            'binary_regex did not match "0" when it should have.'
        )

    def test_empty(self):
        matches = re.fullmatch(binary_regex, '')
        self.assertIsNone(
            matches,
            'binary_regex matched an empty string when it should not have.'
        )

    def test_long(self):
        matches = re.fullmatch(binary_regex, '0110100110')
        self.assertIsNotNone(
            matches,
            'binary_regex did not match "0110100110" when it should have.'
        )

    def test_not_binary(self):
        matches = re.fullmatch(binary_regex, '40')
        self.assertIsNone(
            matches,
            'binary_regex matched "40" when it should not have.'
        )


class EvaluateHawaiian(TestCase):

    def test_aloha(self):
        matches = re.fullmatch(hawaiian_regex, 'Aloha')
        self.assertIsNotNone(
            matches,
            'hawaiian_regex did not match "Aloha" when it should have.'
        )

    def test_empty(self):
        matches = re.fullmatch(hawaiian_regex, '')
        self.assertIsNone(
            matches,
            'hawaiian_regex matched an empty string when it should not have.'
        )

    def test_long(self):
        matches = re.fullmatch(hawaiian_regex, 'humuhumunukunukuapua\'a')
        self.assertIsNotNone(
            matches,
            'hawaiian_regex did not match "a" when it should have.'
        )

    def test_caps_in_middle(self):
        matches = re.fullmatch(hawaiian_regex, 'HolOholo')
        self.assertIsNone(
            matches,
            'hawaiian_regex matched "HolOholo" when it should not have.'
        )

    def test_not_hawaiian(self):
        matches = re.fullmatch(hawaiian_regex, 'Bacana')
        self.assertIsNone(
            matches,
            'hawaiian_regex matched "Bacana" when it should not have.'
        )


class EvaluateTelephone(TestCase):
    def test_valid(self):
        matches = re.fullmatch(telephone_regex, '123-456-7890')
        self.assertIsNotNone(
            matches,
            'telephone_regex did not match "7890" when it should have.'
        )

    def test_too_short(self):
        matches = re.fullmatch(telephone_regex, '12-456-7890')
        self.assertIsNone(
            matches,
            'telephone_regex matched "7890" when it should not have.'
        )

    def test_too_long(self):
        matches = re.fullmatch(telephone_regex, '1234-456-7890')
        self.assertIsNone(
            matches,
            'telephone_regex matched "7890" when it should not have.'
        )

    def test_not_numbers(self):
        matches = re.fullmatch(telephone_regex, 'abc-def-hijk')
        self.assertIsNone(
            matches,
            'telephone_regex matched "hijk" when it should not have.'
        )

    def test_no_dashes(self):
        matches = re.fullmatch(telephone_regex, '1234567890')
        self.assertIsNone(
            matches,
            'telephone_regex matched "1234567890" when it should not have.'
        )


class EvaluateSixPlusNoA(TestCase):
    def test_regex(self):
        matches = re.fullmatch(six_plus_no_a_regex, 'Regular Expressions')
        self.assertIsNone(
            matches,
            'six_plus_no_a_regex matched "Expressions" when it should not have.'
        )

    def test_python(self):
        matches = re.fullmatch(six_plus_no_a_regex, 'Python')
        self.assertIsNotNone(
            matches,
            'six_plus_no_a_regex did not match "Python" when it should have.'
        )

    def test_unique_newyork(self):
        matches = re.fullmatch(six_plus_no_a_regex, 'unique New York')
        self.assertIsNotNone(
            matches,
            'six_plus_no_a_regex did not match "York" when it should have.'
        )

    def test_aloha(self):
        matches = re.fullmatch(six_plus_no_a_regex, 'ALOHA')
        self.assertIsNone(
            matches,
            'six_plus_no_a_regex matched "ALOHA" when it should not have.'
        )

File: exercises/javascript/2_collections_and_ranges/evaluate.js
import {
  binaryRegex as rawBinaryRegex,
  hawaiianRegex as rawHawaiianRegex,
  telephoneRegex as rawTelephoneRegex,
  sixPlusNoARegex as rawSixPlusNoARegex,
} from '.';

// adding string boundaries so regexes behave as expected
const binaryRegex = RegExp(`^${rawBinaryRegex.source}$`);
const hawaiianRegex = RegExp(`^${rawHawaiianRegex.source}$`);
const telephoneRegex = RegExp(`^${rawTelephoneRegex.source}$`);
const sixPlusNoARegex = RegExp(`^${rawSixPlusNoARegex.source}$`);

describe('EvaluateBinary', function () {
  it('matches zero', function () {
    expect(binaryRegex.test('0')).toBe(true);
  });
  it('does not match empty', function () {
    expect(binaryRegex.test('')).toBe(false);
  });
  it('matches long', function () {
    expect(binaryRegex.test('0110100110')).toBe(true);
  });
  it('does not match not_binary', function () {
    expect(binaryRegex.test('40')).toBe(false);
  });
});

describe('EvaluateHawaiian', function () {
  it('matches aloha', function () {
    expect(hawaiianRegex.test('Aloha')).toBe(true);
  });
  it('does not match empty', function () {
    expect(hawaiianRegex.test('')).toBe(false);
  });
  it('matches long', function () {
    expect(hawaiianRegex.test("humuhumunukunukuapua'a")).toBe(true);
  });
  it('does not match caps_in_middle', function () {
    expect(hawaiianRegex.test('HolOholo')).toBe(false);
  });
  it('does not match not_hawaiian', function () {
    expect(hawaiianRegex.test('Bacana')).toBe(false);
  });
});

describe('EvaluateTelephone', function () {
  it('matches valid', function () {
    expect(telephoneRegex.test('123-456-7890')).toBe(true);
  });
  it('does not match too_short', function () {
    expect(telephoneRegex.test('12-456-7890')).toBe(false);
  });
  it('does not match too_long', function () {
    expect(telephoneRegex.test('1234-456-7890')).toBe(false);
  });
  it('does not match not_numbers', function () {
    expect(telephoneRegex.test('abc-def-hijk')).toBe(false);
  });
  it('does not match no_dashes', function () {
    expect(telephoneRegex.test('1234567890')).toBe(false);
  });
});

describe('EvaluateSixPlusNoA', function () {
  it('does not match regex', function () {
    expect(sixPlusNoARegex.test('Regular Expressions')).toBe(false);
  });
  it('matches python', function () {
    expect(sixPlusNoARegex.test('Python')).toBe(true);
  });
  it('matches unique_newyork', function () {
    expect(sixPlusNoARegex.test('unique New York')).toBe(true);
  });
  it('does not match aloha', function () {
    expect(sixPlusNoARegex.test('ALOHA')).toBe(false);
  });
});
